internal/transport/http_util.go

```diff
@@ -335,10 +335,12 @@ func (w *bufWriter) Write(b []byte) (n int, err error) {
 		w.offset += nn
 		n += nn
 		if w.offset >= w.batchSize {
-			err = w.flushKeepBuffer()
```

Comment:
If would be great if we can make `flushKeepBuffer` return `(int, error)` as part of this PR itself. That way, we can make the change here to:

```golang
if w.offset >= w.batchsize {
  if nn, err = w.flushKeepBuffer(); err != nil {
    return n+nn, err
  }
}
```

And we can also modify the test added as part of this change to also verify the returned count of bytes.

But if you want to do it as a follow-up PR, that is fine too. Thanks.

Reply:
I'll create a separate PR for that.